src/test/java/org/broadinstitute/hellbender/tools/copynumber/PreprocessIntervalsIntegrationTest.java

```diff
@@ -0,0 +1,146 @@
+
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import htsjdk.samtools.util.IntervalList;
+import htsjdk.samtools.util.Interval;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.io.File;
+
+public final class PreprocessIntervalsIntegrationTest extends CommandLineProgramTest {
+    private static final File REFERENCE = new File(b37_reference_20_21);
+
+    @DataProvider(name = "intervalInputsFromCommandLine")
+    public final Object[][] testData() {
+        // Test for separate intervals
+        final int binLengthSeparateIntervalTest = 10_000;
+        final int paddingSeparateIntervalTest = 0;
+        final String[] outputFileNameSeparateIntervalTest =  {"GATK-preprocess-intervals-test", ".tmp"};
+        final ArrayList<Interval> inputIntervalsSeparateIntervalTest = new ArrayList<Interval>();
+        inputIntervalsSeparateIntervalTest.add(new Interval("20", 3_000, 20_000));
+        inputIntervalsSeparateIntervalTest.add(new Interval("20", 200, 1_900));
+        final ArrayList<Interval> expectedBinsSeparateIntervalTest = new ArrayList<Interval>();
+        expectedBinsSeparateIntervalTest.add(new Interval("20", 200, 1_900));
+        expectedBinsSeparateIntervalTest.add(new Interval("20", 3_000, 12_999));
+        expectedBinsSeparateIntervalTest.add(new Interval("20", 13_000, 20_000));
+
+        // Test for overlapping intervals
+        final int binLengthOverlappingIntervalTest = 10_000;
+        final int paddingOverlappingIntervalTest = 500;
+        final String[] outputFileNameOverlappingIntervalTest =  {"GATK-preprocess-intervals-test", ".tmp"};
+        final ArrayList<Interval> inputIntervalsOverlappingIntervalTest = new ArrayList<Interval>();
+        inputIntervalsOverlappingIntervalTest.add(new Interval("20", 3_000, 20_000));
+        inputIntervalsOverlappingIntervalTest.add(new Interval("20", 200, 2_100));
+        final ArrayList<Interval> expectedBinsOverlappingIntervalTest = new ArrayList<Interval>();
+        expectedBinsOverlappingIntervalTest.add(new Interval("20", 1, 10_000));
+        expectedBinsOverlappingIntervalTest.add(new Interval("20", 10_001, 20_000));
+        expectedBinsOverlappingIntervalTest.add(new Interval("20", 20_001, 20_500));
+
+        // Test for intervals reaching the ends of the contigs
+        final int binLengthEdgeIntervalTest = 10_000;
+        final int paddingEdgeIntervalTest = 500;
+        final String[] outputFileNameEdgeIntervalTest =  {"GATK-preprocess-intervals-test", ".tmp"};
+        final ArrayList<Interval> inputIntervalsEdgeIntervalTest = new ArrayList<Interval>();
+        inputIntervalsEdgeIntervalTest.add(new Interval("20", 3000, 20000));
+        inputIntervalsEdgeIntervalTest.add(new Interval("20", 63025220, 63025520));
```

Comment:
This is nitpicky, but if you're going to use the `000_000` notation, perhaps use it everywhere consistently?

Reply:
Thanks, I changed it everywhere.